xapian-bindings/java/java.i

```diff
@@ -189,6 +189,64 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%typemap(javainterfaces) Xapian::MSetIterator "Iterable<MSetIterator.MatchDescriptor>";
+%typemap(javaimports) Xapian::MSetIterator "import java.util.Iterator;"
+%typemap(javacode) MSetIterator %{
+  public Iterator<MatchDescriptor> iterator() {
+      Iterator<MatchDescriptor> iterator = new Iterator <MatchDescriptor>() {
+          @Override
+          public boolean hasNext() {
+              return MSetIterator.this.hasNext();
+          }
+          @Override
+          public MatchDescriptor next() {
+              MSetIterator thisIterator = MSetIterator.this;
+              long rank = thisIterator.getRank();
+              long docId = thisIterator.getDocId();
+              double weight = thisIterator.getWeight();
+              int percent = thisIterator.getPercent();
+              Document doc = thisIterator.getDocument();
+              MatchDescriptor descriptor = new MatchDescriptor(docId, rank, percent, weight, doc);
+	      thisIterator.next();
+              return descriptor;            
```

Comment:
There's an indentation difference here (it'll look OK in an editor, but in the patch viewer the leading `+` means it's visible).  Looks like line 210 has a tab, but the other lines only use spaces.  It's better to use consistent indentation.